### ISSUE TITLE
Add checks to block PRs merging changes which include 'mystagingwebsite.com'

### ISSUE THREAD

**Issue by dsas** [Automattic/themes#8253]
I want to prevent #8252 from reoccurring - people committing links to their development site, or embedding resources from them. 

Arguably we should block all external links outside of a small allowlist...

**Comment by alaczek** [comment]
Yes, plus 100! We might also want to check for local links, e.g. `http://localhost:xxxx/`
